docs/datatypes/matrices.md

```diff
@@ -331,6 +331,44 @@ method `.set()`, the matrix will be resized. By default, new items will be
 initialized with zero, but it is possible to specify an alternative value using
 the optional third argument `defaultValue`.
 
+## Advanced Indexing
+
+It is also possible to use an array of booleans for index. This allows some advanced 
+capabilities as the array can come from conditions of other functions it allows for 
+filtering, replacing many values and setting specific values.
+
+```js
+const q = [1, 2, 3, 4]
+math.subset(q, math.index([true, false, true, false]))      // Array [1, 3]
+// making a filter
```

Comment:
maybe skip a line after the first example to make it easier to see that the comments describe the following item, not the preceding one? And/or maybe a colon at the end of the comments would help indicate that.